### PR TITLE
[5249] Bulk QTS import validation

### PR DESCRIPTION
**Pull request by d-a-v-e**
### Context
Adds CSV import validation to both the file and rows. See  https://docs.google.com/spreadsheets/d/10rr-UcbAU41aMg1rbWF_fNiU0qESwH5RxqjFVCLTuHo/edit#gid=0 for which validations are present

### CODE REVIEW

**Review by AbigailMcP — COMMENTED** [review]
This is great! This is such a good base for us to continue with. I like the separate validation files, it's really easy to read and see what's going on.

Just a couple of comments inline to address.

Also I noticed you've added files to `spec/fixtures/files` but (v confusingly!) all the rest (except one) of our fixtures live in `spec/support/fixtures`. Would you mind shifting them to the same place?

File: app/services/bulk_update/recommendations_uploads/validate_csv_row.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module BulkUpdate
+  module RecommendationsUploads
+    class ValidateCsvRow
+      # rubocop:disable Style/OpenStructUse
```
[the remark this answers is not in the document]
Comment (d-a-v-e) [reply app/services/bulk_update/recommendations_uploads/validate_csv_row.rb:6]: I've removed and created a dynamic class `BulkUpdate::RecommendationsUploads::Row` which uses the headers defined in `Reports::BulkRecommendReport` as the source of expected headers 👍

File: app/lib/error_summary_show_all_messages_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class ErrorSummaryShowAllMessagesPresenter
```
[the remark this answers is not in the document]
Comment (d-a-v-e) [reply app/lib/error_summary_show_all_messages_presenter.rb:3]: yeah I was looking at the view folder and the lib and flipped a coin 🤣 

I think I'll create a presenter module as it's a pretty normal thing to have.

**Review by AbigailMcP — APPROVED** [review]
Nice work!

**Pull request merged**